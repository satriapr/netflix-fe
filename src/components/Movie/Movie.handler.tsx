import { useState } from 'react'

// Hooks
const Handler = () => {
  const [isHover, setIsHover] = useState<boolean>(false)

  const handleEnter = () => {
    setIsHover(true)
  }

  const handleLeave = () => {
    setIsHover(false)
  }

  return { isHover, handleEnter, handleLeave }
}

export default Handler
