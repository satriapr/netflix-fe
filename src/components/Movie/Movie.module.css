.movieTitle {
  position: absolute;
  top: 40%;
  left: 50%;
  margin-right: -50%;
  transform: translate(-50%, -50%);
  text-align: center;
  width: 100%;
  white-space: initial;
  color: #fff;
  visibility: hidden;
  opacity: 0;
  transition: opacity 0.2s, visibility 0.2s;
}

.movieCard {
  position: relative;
  width: 170px;
  height: 250px;
  transition: all 0.7s;
  background: black;
}

.movieCard:hover {
  width: 190px;
  height: 280px;
}

.movieCard:hover .movieTitle {
  visibility: visible;
  opacity: 1;
}

.movieCard:hover img {
  width: 190px;
  height: 280px;
  vertical-align: -15px;
  opacity: 0.3;
}

@media only screen and (max-width: 900px) {
  .movieCard,
  img {
    width: 140px;
    height: 180px;
  }

  .movieCard:hover,
  .movieCard:hover img {
    width: 160px;
    height: 210px;
  }
}

@media only screen and (max-width: 690px) {
  .movieCard,
  img {
    width: 100px;
    height: 140px;
  }

  .movieCard:hover,
  .movieCard:hover img {
    width: 120px;
    height: 180px;
    vertical-align: -20px;
  }

  .movieTitle {
    font-size: 14px;
  }
}

@media only screen and (max-width: 350px) {
  .movieCard,
  img {
    width: 90px;
    height: 130px;
  }

  .movieCard:hover,
  .movieCard:hover img {
    width: 110px;
    height: 120px;
  }
}