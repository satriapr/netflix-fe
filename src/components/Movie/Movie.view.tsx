import React from 'react'
import { Link } from 'react-router-dom'

import { IMovieRes } from '../../models'
import MovieHandler from './Movie.handler'
import styles from './Movie.module.css'

const Movie = ({ Title, imdbID, Poster }: IMovieRes) => {
  const handler = MovieHandler()
  const { handleEnter, handleLeave, isHover } = handler

  return (
    <Link to={`/${imdbID}`}>
      <div
        className={styles.movieCard}
        onMouseEnter={handleEnter}
        onMouseLeave={handleLeave}
      >
        <img src={Poster} alt={Title} />
        {isHover && <h3 className={styles.movieTitle}>{Title}</h3>}
      </div>
    </Link>
  )
}

export default Movie
