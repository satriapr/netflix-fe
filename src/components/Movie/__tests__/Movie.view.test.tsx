import Movie from '../Movie.view'
import { testSnapshots } from '../../../utils/test.util'

describe('Movie Component', () => {
  testSnapshots(Movie, [
    {
      description: 'Movie props',
      props: {
        Title: 'Running Man',
        imdbID: '123123',
        Poster: ''
      }
    }
  ])
})
