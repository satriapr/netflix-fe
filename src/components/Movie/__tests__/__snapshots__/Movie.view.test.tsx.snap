// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Component snapshots Movie props 1`] = `
<Link
  to="/123123"
>
  <div
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <img
      alt="Running Man"
      src=""
    />
  </div>
</Link>
`;
